x/poolmanager/router.go

```diff
@@ -273,67 +273,71 @@ func (k Keeper) MultihopEstimateOutGivenExactAmountIn(
 	return tokenOutAmount, err
 }
 
-// MultihopSwapExactAmountOut defines the output denom and output amount for the last pool.
-// Calculation starts by providing the tokenOutAmount of the final pool to calculate the required tokenInAmount
-// the calculated tokenInAmount is used as defined tokenOutAmount of the previous pool, calculating in reverse order of the swap
-// Transaction succeeds if the calculated tokenInAmount of the first pool is less than the defined tokenInMaxAmount defined.
+// RouteExactAmountIn processes a swap along the given route using the swap function corresponding
+// to poolID's pool type. This function is responsible for computing the optimal input amount
+// for a given output amount when swapping tokens, taking into account the current price of the
+// tokens in the pool and any slippage.
+// Transaction succeeds if final amount out is greater than tokenOutMinAmount defined
+// and no errors are encountered along the way.
```

Comment:
This is the wrong godoc, its talking about in logic when it should be talking about out logic